### DISCUSSION TITLE
I need help running OpenModelica in a Docker container

### DISCUSSION THREAD

**Discussion by vtnate** [OpenModelica/OpenModelica#10643] [Q&A] [answered]
I'd like to use v1.21 from https://openmodelica.org/download/docker/ to run OpenModelica on the command line in a Docker container, and I have a couple questions:

1. The text says v1.21 is available, but when I click the [Docker Hub link](https://hub.docker.com/r/openmodelica/openmodelica/tags ) 1.20 is the newest one available. Is 1.21 available somewhere else?
2. The text says -minimal can run `omc` from the command line. Is `OMSimulator` available in the minimal image? If not, which image should I use to run `OMSimulator` on the command line?

**Comment by AnHeuermann** [comment, marked as answer]
Looks like the v1.20.0 images are not yet uploaded to DockerHub, but the the nightly version image `openmodelica/openmodelica:nightly` is available.

To see if `OMSimulator` is provided by one of the images you can try to run it:

```bash
$ docker run --rm -it openmodelica/openmodelica:v1.20.0-minimal OMSimulator --version
docker: Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "OMSimulator": executable file not found in $PATH: unknown.
```

So it is not included in the minimal image, but it is available in `openmodelica/openmodelica:v1.20.0-gui`.

You can create your own Dockerfile and follow the install instructions from [openmodelica.org/download/download-linux](https://openmodelica.org/download/download-linux/) to create an image with all the tools you need.
This should look something like [OpenModelicaBuildScripts Dockerfile.nightly](https://github.com/OpenModelica/OpenModelicaBuildScripts/blob/master/docker/Dockerfile.nightly).

**Reply by AnHeuermann** [reply]
The OpenModelica v1.21 images `openmodelica/openmodelica:v1.21.0-minimal`, `openmodelica/openmodelica:v1.21.0-ompython` and `openmodelica/openmodelica:v1.21.0-gui` are now available as well: https://hub.docker.com/r/openmodelica/openmodelica/tags?page=1&name=v1.21
